maliput/examples/01_road_geometry.rs

```diff
@@ -55,4 +55,5 @@ fn main() {
     for lane in lanes {
         println!("\tlane id: {}", lane.id());
     }
+    Ok::<(), Box<maliput::common::MaliputError>>(())
```

Comment:
Why is this one different?

Reply:
this main was returning `Result<(), impl std::error::Error>` so at compile time it didn't have the info about the possible error that it might return, so this couldn't be ellided.
I switched to `dyn std::error::Error` as no need to have a different one here, so the return type now is Ok(())